sqlalchemy_ibmi/base.py

```diff
@@ -748,51 +748,13 @@ def get_isolation_level(self, dbapi_conn):
     def create_connect_args(self, url):
         opts = url.translate_connect_args(username='user')
         opts.update(url.query)
-        keys = opts
-
-        connect_args = {}
-        for param in ('ansi', 'unicode_results', 'autocommit'):
-            if param in keys:
-                connect_args[param] = util.asbool(keys.pop(param))
-
-        if 'odbc_connect' in keys:
-            connectors = [urllib.parse.unquote_plus(keys.pop('odbc_connect'))]
+        allowed_opts = {'host', 'user', 'password',
```

Comment:
We need to add database to this.

Also, we may need to handle port separately. We should ensure that it is not specified. Not sure how translate_connect_args handles that. Is it not in the dictionary? Is it `None`/empty string? Once we've validated it's not set then we should remove it from the dictionary so it doesn't disrupt the below check. If translate_connect_args doesn't add port to the dictionary if it wasn't specified, then I think we can leave things as they are.

Reply:
Port will be passed as None, added a commit to handle that.